qemu/tests/cpu_model_inter_generation.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import cpu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Boot latest cpu model on old host
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def start_with_model(test_model):
+        """
+        Start vm with tested model
+        :param test_model: The model been tested
+        """
+        vm = None
+        params['cpu_model'] = test_model
+        logging.info('Start vm with cpu model %s' % test_model)
+        try:
+            env_process.preprocess_vm(test, params, env, vm_name)
+            vm = env.get_vm(vm_name)
+            output = vm.process.get_output()
+            if warning_text not in output:
+                test.fail("Qemu should output warning for lack flags"
+                          " while it does not.")
+        except Exception as e:
+            if boot_expected == 'no':
+                logging.info('Expect vm boot up failed when enforce is set.')
+                if warning_text not in str(e):
+                    raise
+            else:
+                raise
+        else:
+            if boot_expected == 'no':
+                test.fail('The vm should not boot successfully'
+                          ' when cpu enforce mode is on')
+        finally:
+            if vm and vm.is_alive():
+                vm.destroy()
+
+    fd = open("/proc/cpuinfo")
+    cpu_info = fd.read()
+    fd.close()
+    vendor = cpu.get_cpu_vendor(cpu_info)
+    latest_cpu_model = cpu.CPU_TYPES.get(vendor)[0]
```

Comment:
Hi, we need to check if the `latest_cpu_model` is supported by current qemu version, if no, we need to pick up the next cpu model in cpu_types.

Reply:
@nanliu-r Updated, please help check, thanks.